API/chat/forms.py

```diff
@@ -51,3 +54,67 @@ def save(self, message):
         message.typing = self.cleaned_data.get('typing', False)
 
         message.save()
+
+
+class SessionForm(AuthenticationForm):
+    username = forms.CharField(max_length=20)
+    password = forms.CharField(max_length=32, widget=forms.PasswordInput)
+
+    def clean(self):
+        username = self.cleaned_data.get('username')
+        password = self.cleaned_data.get('password')
+
+        if username is None:
+            raise forms.ValidationError(
+                'invalid_username',
+                code='invalid_username'
+           )
+
+        try:
+            user = User.objects.get(username=username)
+
+            if user.has_usable_password():
+                if password is None:
+                    raise forms.ValidationError(
+                        'password_required',
+                        code='password_required'
+                    )
+
+                auth = authenticate(username=username, password=password)
+
+                if auth is None:
+                    raise forms.ValidationError(
+                        'wrong_password',
+                        code='wrong_password'
+                    )
+
+                return "Authenticate"
+
+            raise forms.ValidationError(
+                'username_reserved',
+                code='username_reserved'
+            )
```

Comment:
Check `last_used` before raising. If `last_used` is old, allow the user to reuse this username.